CDCgov/prime-reportstream
# Add Mars OTC ELR scenarios to E2E UP Test

This PR adds end 2 end testing scenarios for Mars OTC ELR reports moving through the UP. 
These scenarios test when reports with the Mars OTC ELR topic
- Are sent in HL7 and received as HL7 and FHIR. 
- Are sent in FHIR and received as FHIR.
- When invalid HL7 items are present in the report, these are filtered out.

Test Steps:
1. Run `./gradlew testend2endup`
2. Verify that all scenarios pass.
    1. For scenario "Sending HL7 Report, Receiving HL7/FHIR (mars-otc-elr); Invalid HL7 Items Filtered Out":
        * A FHIR report should be sent to MARS_OTC_ELR_FHIR_B_E2E 
        * An HL7 report should be sent to MARS_OTC_ELR 
        * Both sent reports consist only of one item
        * One item has been filtered out with the message `"Item 1 in the report was not valid. Reason: HL7 was not valid at OBX[1]-19[1].1"`
    2. For scenario "Sending FHIR Report, Receiving FHIR (mars-otc-elr)":
        * A FHIR report should be sent to MARS_OTC_ELR_FHIR_A_E2E 

## Changes
- Added two MARS OTC ELR scenarios in the UP end to end test
- Modified test logging in a few places
- Added MARS OTC ELR Senders/Receivers

## Checklist

### Testing

## Linked Issues
- Fixes https://github.com/CDCgov/prime-reportstream/issues/14336
